lib/quicly.c

```diff
@@ -2464,22 +2464,23 @@ static ssize_t round_send_window(ssize_t window)
  */
 static size_t calc_send_window(quicly_conn_t *conn, size_t min_bytes_to_send, int restrict_sending)
 {
-    /* If address is unvalidated, limit sending to 3x bytes received */
-    if (!conn->super.peer.address_validation.validated) {
-        uint64_t total = conn->super.stats.num_bytes.received * 3;
-        if (conn->super.stats.num_bytes.sent + MIN_SEND_WINDOW <= total)
-            return total - conn->super.stats.num_bytes.sent;
-        return 0;
-    }
-
-    /* Validated address. Ensure there's enough window to send minimum number of packets */
     uint64_t window = 0;
     if (!restrict_sending && conn->egress.cc.cwnd > conn->egress.sentmap.bytes_in_flight + min_bytes_to_send)
         window = conn->egress.cc.cwnd - conn->egress.sentmap.bytes_in_flight;
     if (window < MIN_SEND_WINDOW)
         window = 0;
-    if (window < min_bytes_to_send)
-        window = min_bytes_to_send;
+    /* If address is validated, ensure there's enough window to send minimum number of packets */
+    if (conn->super.peer.address_validation.validated) {
+        if (window < min_bytes_to_send)
+            window = min_bytes_to_send;
+        return window;
+    }
```

Comment:
I am not sure if it is correct to added the surrounding `if`.

Previously, our CC logic is `if (!PTO) calc_window_using_cwnd else use_min_bytes`. I think that needs to be retained, otherwise, we would not be sending a single byte in case of PTO, even when the number of bytes previously been sent is below the 3x limit.